Added check for doStepIf types

campos-ddc opened the pull request.

campos-ddc committed:
Added check for doStepIf types

mariangemarcano reviewed (commented):
@campos-ddc I guess this fixes a bug ? how did you hit it?

campos-ddc commented:
@mariangemarcano There was a builder passing a list containing another list to `doStepIf`. The code only handles one level, so I added a check to guarantee that (and fixed my builder)

Usually I wouldn't add type checking in Python, but this kind of error only shows up in runtime (`doStepIf` is only executed on the slaves), so I need this to identify errors during load :)

campos-ddc committed:
Added test for doStepIf type checking
